pkg/kubevirt/actuator.go

```diff
@@ -0,0 +1,275 @@
+/*
+Copyright 2019 The OpenShift Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package kubevirt
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/openshift/cloud-credential-operator/pkg/operator/constants"
+	actuatoriface "github.com/openshift/cloud-credential-operator/pkg/operator/credentialsrequest/actuator"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	minterv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	kubernetesErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type KubevirtActuator struct {
+	Client client.Client
+	Codec  *minterv1.ProviderCodec
+}
+
+const (
+	KubevirtCredentialsSecretKey = "kubeconfig"
+)
+// NewActuator creates a new Kubevirt actuator.
+func NewActuator(client client.Client) (*KubevirtActuator, error) {
+	codec, err := minterv1.NewCodec()
+	if err != nil {
+		log.WithError(err).Error("error creating Kubevirt codec")
+		return nil, fmt.Errorf("error creating Kubevirt codec: %v", err)
+	}
+
+	return &KubevirtActuator{
+		Codec:  codec,
+		Client: client,
+	}, nil
+}
+
+// Exists checks if the credentials currently exist.
+// TODO: in the future validate the expiration of the credentials
+func (a *KubevirtActuator) Exists(ctx context.Context, cr *minterv1.CredentialsRequest) (bool, error) {
```

Comment:
Don't you need to validate that the value inside the secret is as expected?

Reply:
The "getSecret" function validate that the secret contains the required key in the data
Moreover, in the other actoators there isn't such a validation